### PR TITLE
Improve and update E2E tests 

### PR DESCRIPTION
**Pull request by ahmelsayed**
This change:

- increases test timeout to ~5~ 10 minutes
- uses a different namespace for Prometheus for `argo-rollouts.test.ts` and `prometheus.test.ts` to avoid conflict if the tests run at the same time
- reduce `pollingInterval` and `cooldownPeriod` in `azure-pipelines.test.ts`
- add readiness probe on influxdb, and rabbitmq deployment
- use `rabbitmq.conf` and `enabled_plugins` files for configuring rabbitmq instead of env vars since they are [deprecated](https://github.com/docker-library/rabbitmq/pull/467) and cause an error on start up
- add some helper methods for common tasks. 

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [N/A] Tests have been added
- [NA] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [N/A] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [N/A] Changelog has been updated

### CODE REVIEW

File: pkg/scaling/scale_handler.go [hunk 1]
```diff
@@ -6,21 +6,20 @@ import (
...
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/record"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-
-	kedav1alpha1 "github.com/kedacore/keda/v2/api/v1alpha1"
-	"github.com/kedacore/keda/v2/pkg/scalers"
-	"github.com/kedacore/keda/v2/pkg/scaling/executor"
-	"github.com/kedacore/keda/v2/pkg/scaling/resolver"
```
Comment (zroubalik) [inline pkg/scaling/scale_handler.go:23]: I'd actually prefer if we keep the `github.com/kedacore/keda` related imports in the separate section below general imports.  It's just my ocd, sorry 😄

File: pkg/scaling/scale_handler.go [hunk 2]
```diff
@@ -202,11 +201,20 @@ func (h *scaleHandler) checkScalers(ctx context.Context, scalableObject interfac
 	defer scalingMutex.Unlock()
 	switch obj := scalableObject.(type) {
 	case *kedav1alpha1.ScaledObject:
+		err = h.client.Get(ctx, types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace}, obj)
```
Comment (zroubalik) [inline pkg/scaling/scale_handler.go:204]: What is the reason for this change?

File: pkg/scaling/scale_handler.go [hunk 3]
```diff
@@ -202,11 +201,20 @@ func (h *scaleHandler) checkScalers(ctx context.Context, scalableObject interfac
...
+			return
+		}
 		isActive, isError := h.isScaledObjectActive(ctx, scalers, obj)
 		h.scaleExecutor.RequestScale(ctx, obj, isActive, isError)
 	case *kedav1alpha1.ScaledJob:
-		scaledJob := scalableObject.(*kedav1alpha1.ScaledJob)
-		isActive, scaleTo, maxScale := h.isScaledJobActive(ctx, scalers, scaledJob)
+		err = h.client.Get(ctx, types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace}, obj)
+		if err != nil {
+			h.logger.Error(err, "Error getting scaledOJob", "object", scalableObject)
```
Comment (zroubalik) [inline pkg/scaling/scale_handler.go:214]:
```suggestion
			h.logger.Error(err, "Error getting scaledJob", "object", scalableObject)
```

**Review by zroubalik — COMMENTED** [review]
This is great! I have just minor nits in `scale_handler.go`.

File: pkg/scaling/scale_handler.go (hunk 2 above)
Comment (ahmelsayed) [reply pkg/scaling/scale_handler.go:204]: It was to make sure we refresh the scaledObject/scaledJob object before we run the scalers check loop on them. I was patching the status during tests and noticed we still use the old values when checking things like lastActiveTime.

Comment (zroubalik) [reply pkg/scaling/scale_handler.go:206]: Makes sense, that's interesting finding, thanks!

**Review by zroubalik — APPROVED** [review]
LGTM, that's great
